### PR TITLE
Improve doc for obb.ts and intersect.ts

### PR DESCRIPTION
**Pull request by dumganhar**
Re: #13888

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/geometry/intersect.ts [hunk 1]
```diff
@@ -46,9 +46,9 @@ import type { Mesh } from '../../3d';
  * ray-plane intersect detect.
  * @zh
  * 射线与平面的相交性检测。
- * @param ray @zh 射线 @en The ray to test
- * @param plane @zh 平面 @en The plane to test
- * @return @zh 0 或 非0 @en 0 or not 0, 0 indicates there is no intersection
+ * @param ray @zh 射线。 @en The ray to test.
+ * @param plane @zh 平面。 @en The plane to test.
+ * @returns @zh 0 或 非 0，0 表示没有相交。 @en 0 or not 0, 0 indicates there is no intersection.
```
Comment (minggo) [inline cocos/core/geometry/intersect.ts]: I think it is better to change to
`0 if xxx, else not 0`

File: cocos/core/geometry/intersect.ts [hunk 2]
```diff
@@ -69,10 +69,10 @@ const rayPlane = (function () {
...
  * @zh
  * 射线与三角形的相交性检测。
- * @param ray @zh 射线 @en The ray to test
- * @param triangle @zh 三角形 @en The triangle to test
- * @param doubleSided @zh 三角形是否为双面 @en Indicates whether it is double sided
- * @return @zh 0 或 非0 @en 0 or not 0, 0 indicates there is no intersection
+ * @param ray @zh 射线。 @en The ray to test.
+ * @param triangle @zh 三角形。 @en The triangle to test.
+ * @param doubleSided @zh 三角形是否为双面。 @en Indicates whether it is double sided.
+ * @returns @zh 0 或 非 0，0 表示没有相交。@en 0 or not 0, 0 indicates there is no intersection.
```
Comment (minggo) [inline cocos/core/geometry/intersect.ts:75]: The same as above.

File: cocos/core/geometry/intersect.ts [hunk 3]
```diff
@@ -69,10 +69,10 @@ const rayPlane = (function () {
  * ray-triangle intersect detect.
  * @zh
  * 射线与三角形的相交性检测。
- * @param ray @zh 射线 @en The ray to test
- * @param triangle @zh 三角形 @en The triangle to test
- * @param doubleSided @zh 三角形是否为双面 @en Indicates whether it is double sided
- * @return @zh 0 或 非0 @en 0 or not 0, 0 indicates there is no intersection
+ * @param ray @zh 射线。 @en The ray to test.
+ * @param triangle @zh 三角形。 @en The triangle to test.
```
Comment (minggo) [inline cocos/core/geometry/intersect.ts]: 三角形 -> 用来测试的三角形

File: cocos/core/geometry/obb.ts [hunk 4]
```diff
@@ -167,7 +167,7 @@ export class OBB {
 
     /**
      * @en
-     * Center point of the OBB.
+     * The center point of the OBB in local coordinate.
```
Comment (minggo) [inline cocos/core/geometry/obb.ts]: of the OBB -> of an OBB

**Review by minggo — APPROVED** [review]

**Comment by dumganhar** [comment]
@minggo Updated according the comments.

**Pull request merged**